### ISSUE TITLE
Android build issue

### ISSUE THREAD

**Issue by henrymoulton** [bugsnag/bugsnag-js#1550]
### Describe the bug

When doing Android builds I'm getting:

```
> Task :app:bundleStagingReleaseJsAndAssets FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:bundleStagingReleaseJsAndAssets'.
> kotlin.io.FileAlreadyExistsException: /Users/henrymoulton/Code/app/android/app/build/generated/assets/react/staging/release/index.android.bundle -> /Users/henrymoulton/Code/app/android/app/build/intermediates/bugsnag/rescued/Staging-release.android.bundle: The destination file already exists.
```

Running `rm -rf android/app/build` does the trick but I'm losing some caching ability I suspect.

It specifically mentions `/intermediates/bugsnag/rescued/Staging-release.android.bundle:` so that's why I've reported here.

### Steps to reproduce
1. Run an android build
2. Run another android build without running `./gradlew clean` or `rm -rf android/app/build`

### Environment
* Bugsnag version: 7.10.5

**Comment by mattdyoung** [comment]
Closing as we haven't been able to reproduce, but if you can share an example of how to reproduce the issue we'd be happy to re-open and investigate further.

**Comment by rikur** [comment]
This happens to me after every build. I have to manually delete the `./android/app/build/intermediates/bugsnag/rescued/Release.android.bundle` file to be able to build again.

**Comment by rikur** [comment]
Can't share all the files, but the relevant lines in `./android/app/build.gradle` are:

```
apply from: "../../node_modules/react-native/react.gradle"
apply plugin: "com.bugsnag.android.gradle"
```

```
bugsnag {
  uploadReactNativeMappings = true
  uploadJvmMappings = true
  uploadNdkMappings = true
  uploadNdkUnityLibraryMappings = false
  reportBuilds = true
  retryCount = 5
  failOnUploadError = true
}
```

**Comment by luke-belton** [comment]
Hi @rikur - unfortunately I've still been unable to reproduce the behaviour described. I've created a new app with `npx react-native init` and added `@bugsnag/react-native` `v7.10.5`, with the same lines you shared above above in my app's `build.gradle`.

If you're able to provide a cut down reproduction example we'd be happy to look into this for you 👍

**Comment by rikur** [comment]
Thank you for looking into this. I'll keep you updated on what I find.

Here's a stacktrace:
```
Caused by: kotlin.io.FileAlreadyExistsException: project/android/app/build/generated/assets/react/release/index.android.bundle -> project/android/app/build/intermediates/bugsnag/rescued/Release.android.bundle: The destination file already exists.
        at kotlin.io.FilesKt__UtilsKt.copyTo(Utils.kt:224)
        at kotlin.io.FilesKt__UtilsKt.copyTo$default(Utils.kt:217)
        at com.bugsnag.android.gradle.BugsnagUploadJsSourceMapTask$Companion$rescueReactNativeSourceBundle$1.execute(BugsnagUploadJsSourceMapTask.kt:206)
        at com.bugsnag.android.gradle.BugsnagUploadJsSourceMapTask$Companion$rescueReactNativeSourceBundle$1.execute(BugsnagUploadJsSourceMapTask.kt:152)```

**Comment by rikur** [comment]
I updated all bugsnag libraries to the latest version and the issue persists, with the same stacktrace.

```
info Direct dependencies
├─ @bugsnag/react-native@7.14.0
└─ @bugsnag/source-maps@2.3.0
info All dependencies
├─ @bugsnag/core@7.14.0
├─ @bugsnag/delivery-react-native@7.14.0
├─ @bugsnag/plugin-console-breadcrumbs@7.14.0
├─ @bugsnag/plugin-network-breadcrumbs@7.14.0
├─ @bugsnag/plugin-react-native-client-sync@7.14.0
├─ @bugsnag/plugin-react-native-event-sync@7.14.0
├─ @bugsnag/plugin-react-native-global-error-handler@7.14.0
├─ @bugsnag/plugin-react-native-hermes@7.14.0
├─ @bugsnag/plugin-react-native-session@7.14.0
├─ @bugsnag/plugin-react-native-unhandled-rejection@7.14.0
├─ @bugsnag/plugin-react@7.14.0
├─ @bugsnag/react-native@7.14.0
└─ @bugsnag/source-maps@2.3.0
```

**Comment by luke-belton** [comment]
Hi @rikur - could you let us know what version of the Bugsnag Android Gradle Plugin you're using too please?

**Comment by rikur** [comment]
Hmm, I take that back. Latest version fails with a different error after `./gradlew clean`. Looking into it.

**Comment by rikur** [comment]
I have limited understanding of gradle, I hope this answers your question:

```
+--- project :bugsnag_react-native
|    +--- com.bugsnag:bugsnag-android:5.15.0
|    |    +--- androidx.annotation:annotation:1.1.0 -> 1.2.0
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72 -> 1.4.10
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-common:1.4.10
|    |    |    \--- org.jetbrains:annotations:13.0
|    |    +--- com.bugsnag:bugsnag-android-core:5.15.0
|    |    |    +--- androidx.annotation:annotation:1.1.0 -> 1.2.0
|    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72 -> 1.4.10 (*)
|    |    +--- com.bugsnag:bugsnag-plugin-android-anr:5.15.0
|    |    |    +--- androidx.annotation:annotation:1.1.0 -> 1.2.0
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72 -> 1.4.10 (*)
|    |    |    \--- com.bugsnag:bugsnag-android-core:5.15.0 (*)
|    |    \--- com.bugsnag:bugsnag-plugin-android-ndk:5.15.0
|    |         +--- androidx.annotation:annotation:1.1.0 -> 1.2.0
|    |         +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72 -> 1.4.10 (*)
|    |         \--- com.bugsnag:bugsnag-android-core:5.15.0 (*)
|    \--- com.bugsnag:bugsnag-plugin-react-native:5.15.0
|         +--- androidx.annotation:annotation:1.1.0 -> 1.2.0
|         +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72 -> 1.4.10 (*)
|         \--- com.bugsnag:bugsnag-android-core:5.15.0 (*)
```

Configured with:
```
classpath("com.bugsnag:bugsnag-android-gradle-plugin:5.+")
```

**Comment by rikur** [comment]
I may opt for using the bugsnag cli to upload source maps instead. Thank you again for looking into this :)

**Comment by yousif-bugsnag** [comment]
Hi @rikur, thanks for sharing that.

> Latest version fails with a different error after ./gradlew clean. Looking into it.

Sounds like you're now seeing a different issue after upgrading and running `./gradlew clean` - can you confirm if that's the case?

**Comment by rikur** [comment]
Yes, the different issue seemed to be related to gradle/kotlin version and I don't have the bandwidth to look into it more.

Uploading the source maps via the CLI works great.

**Comment by johnkiely1** [comment]
Thanks for the update, we will close this issue for now as it sounds like you have a workaround and we were not able to reproduce. Happy to reopen if any new information comes to light.

**Comment by henrymoulton** [comment]
> Sounds like you're now seeing a different issue after upgrading and running ./gradlew clean - can you confirm if that's the case?

`rm -rf app/build/intermediates/bugsnag/rescued/filename.bundle` before running the script seems to work

**Comment by henrymoulton** [comment]
```
[15:21:44]: ▸ FAILURE: Build failed with an exception.
[15:21:44]: ▸ * What went wrong:
[15:21:44]: ▸ A problem was found with the configuration of task ':app:uploadBugsnagStaging-releaseSourceMaps' (type 'BugsnagUploadJsSourceMapTaskGradle53Plus').
[15:21:44]: ▸ > File 'path/android/app/build/intermediates/bugsnag/rescued/Staging-release.android.bundle' specified for property 'bundleJsFileProvider' does not exist.
```

New error actually as a result of this so perhaps @rikur was right to disable sourcemaps in gradle and use the CLI.

**Comment by mattdyoung** [comment]
We'd still be interested in trying to determine the root cause of this if you're able to share a cut down reproduction example project which demonstrates the issue, as we've so far been unable to reproduce. Feel free to email to support@bugsnag.com if you prefer.

**Comment by luke-belton** [comment]
Hi @henrymoulton, @rikur  - I've just managed to reproduce the `FileAlreadyExistsException` by enabling Hermes in an example app, which wasn't something we tried initially. Could you confirm that you have Hermes enabled in your projects?

We're going to look into a fix for this - will keep you updated here.

**Comment by henrymoulton** [comment]
Yep using Hermes

**Comment by madandrija** [comment]
I have the same issue, using Hermes as well

**Comment by mattdyoung** [comment]
Thanks for the info. We've scheduled a fix for this early next month.

**Comment by madandrija** [comment]
@mattdyoung - I just tried `7.16.0` and I'm still getting the same error, do you perhaps know when the fix will be in?

**Comment by luke-belton** [comment]
Hi @madandrija - we're working on getting the fix released. We'll let you know here as soon as it's available.

**Comment by johnkiely1** [comment]
Hi all, 

Just letting you know we have now fixed this issue in [v5.8.2](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v5.8.2) of our gradle plugin. 
Let us know if you have any issues once you upgrade to this version.

**Comment by hardworker** [comment]
@johnkiely1 Any plans to cherry-pick this fix to `v7` while it's recommended by docs?

**Comment by bradleyflood** [comment]
Hey @luke-belton will this fix be applied to `v7` too? I'm also experiencing the same.

**Comment by johnkiely1** [comment]
Hi @hardworker, @bradleyflood. 

We are currently investigating porting this fix to v7 to see whats involved. We will let you know of updates as soon as we can.

**Comment by johnkiely1** [comment]
Hi all, 
We've just released this fix in [v7.4.0](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v7.4.0) of the plugin. Do let us know if you see any issues when you upgrade to it.

**Comment by bradleyflood** [comment]
Epic. Thank you so much @johnkiely1  it works! :)
